src/Memoize.jl

```diff
@@ -76,4 +78,18 @@ macro memoize(args...)
     end)
 
 end
+
+function memoize_cache(m::Module, f::Function)
+    fmodulename = which(m, Symbol(f))
+    getproperty(fmodulename, cache_name(f))
```

Comment:
```suggestion
    getproperty(parentmodule(f), cache_name(f))
```
Would that work? Then you can get rid of the `m` argument entirely, and `memoize_cache(f)` becomes a plain function.

It'll fail in certain circumstances (eg. `@memoize Base.sin(::MyNumberType) = ...`) but I don't think there's a clean answer here, because you can already have multiple caches for `Base.sin` if the methods are defined in different modules.

Reply:
Awesome suggestion! :smile: